unit-test failure: TestManifestSuite/TestReconcileKubeProxyExtraArgs

## Bug Report

unit-test flakiness

### Description

### Logs

```
#166 15.06 === RUN   TestManifestSuite
#166 15.06 === RUN   TestManifestSuite/TestReconcileDefaults
#166 15.06 controller-runtime:  k8s.ManifestController: 2021/04/12 21:19:56 controller starting
#166 15.06     manifest_test.go:210: tear down
#166 15.06 controller-runtime:  k8s.ManifestController: 2021/04/12 21:19:56 controller finished
#166 15.06 === RUN   TestManifestSuite/TestReconcileDisableKubeProxy
#166 15.06 controller-runtime:  k8s.ManifestController: 2021/04/12 21:19:56 controller starting
#166 15.06     manifest_test.go:210: tear down
#166 15.06 controller-runtime:  k8s.ManifestController: 2021/04/12 21:19:56 controller finished
#166 15.06 === RUN   TestManifestSuite/TestReconcileKubeProxyExtraArgs
#166 15.06 controller-runtime:  k8s.ManifestController: 2021/04/12 21:19:56 controller starting
#166 15.06     manifest_test.go:198: 
#166 15.06         	Error Trace:	manifest_test.go:198
#166 15.06         	Error:      	"[%!s(*unstructured.Unstructured=&{map[apiVersion:apps/v1 kind:DaemonSet metadata:map[labels:map[k8s-app:kube-proxy tier:node] name:kube-proxy namespace:kube-system] spec:map[selector:map[matchLabels:map[k8s-app:kube-proxy tier:node]] template:map[metadata:map[labels:map[k8s-app:kube-proxy tier:node]] spec:map[containers:[map[command:[/usr/local/bin/kube-proxy --cluster-cidr=10.244.0.0/16 --hostname-override=$(NODE_NAME) --kubeconfig=/etc/kubernetes/kubeconfig --proxy-mode= --conntrack-max-per-core=0 --bind-address="::"] env:[map[name:NODE_NAME valueFrom:map[fieldRef:map[fieldPath:spec.nodeName]]]] image:foo/bar name:kube-proxy securityContext:map[privileged:true] volumeMounts:[map[mountPath:/lib/modules name:lib-modules readOnly:true] map[mountPath:/etc/ssl/certs name:ssl-certs-host readOnly:true] map[mountPath:/etc/kubernetes name:kubeconfig readOnly:true]]]] hostNetwork:true serviceAccountName:kube-proxy tolerations:[map[effect:NoSchedule operator:Exists] map[effect:NoExecute operator:Exists]] volumes:[map[hostPath:map[path:/lib/modules] name:lib-modules] map[hostPath:map[path:/etc/ssl/certs] name:ssl-certs-host] map[configMap:map[name:kubeconfig-in-cluster] name:kubeconfig]]]] updateStrategy:map[rollingUpdate:map[maxUnavailable:1] type:RollingUpdate]]]}) %!s(*unstructured.Unstructured=&{map[apiVersion:v1 kind:ServiceAccount metadata:map[name:kube-proxy namespace:kube-system]]}) %!s(*unstructured.Unstructured=&{map[apiVersion:rbac.authorization.k8s.io/v1 kind:ClusterRoleBinding metadata:map[name:kube-proxy] roleRef:map[apiGroup:rbac.authorization.k8s.io kind:ClusterRole name:system:node-proxier] subjects:[map[kind:ServiceAccount name:kube-proxy namespace:kube-system]]]}) %!s(*unstructured.Unstructured=&{map[apiVersion:apps/v1 kind:DaemonSet metadata:map[labels:map[k8s-app:kube-proxy tier:node] name:kube-proxy namespace:kube-system] spec:map[selector:map[matchLabels:map[k8s-app:kube-proxy tier:node]] template:map[metadata:map[labels:map[k8s-app:kube-proxy tier:node]] spec:map[containers:[map[command:[/usr/local/bin/kube-proxy --cluster-cidr=10.244.0.0/16 --hostname-override=$(NODE_NAME) --kubeconfig=/etc/kubernetes/kubeconfig --proxy-mode= --conntrack-max-per-core=0 --bind-address="::"] env:[map[name:NODE_NAME valueFrom:map[fieldRef:map[fieldPath:spec.nodeName]]]] image:foo/bar name:kube-proxy securityContext:map[privileged:true] volumeMounts:[map[mountPath:/lib/modules name:lib-modules readOnly:true] map[mountPath:/etc/ssl/certs name:ssl-certs-host readOnly:true] map[mountPath:/etc/kubernetes name:kubeconfig readOnly:true]]]] hostNetwork:true serviceAccountName:kube-proxy tolerations:[map[effect:NoSchedule operator:Exists] map[effect:NoExecute operator:Exists]] volumes:[map[hostPath:map[path:/lib/modules] name:lib-modules] map[hostPath:map[path:/etc/ssl/certs] name:ssl-certs-host] map[configMap:map[name:kubeconfig-in-cluster] name:kubeconfig]]]] updateStrategy:map[rollingUpdate:map[maxUnavailable:1] type:RollingUpdate]]]}) %!s(*unstructured.Unstructured=&{map[apiVersion:v1 kind:ServiceAccount metadata:map[name:kube-proxy namespace:kube-system]]}) %!s(*unstructured.Unstructured=&{map[apiVersion:rbac.authorization.k8s.io/v1 kind:ClusterRoleBinding metadata:map[name:kube-proxy] roleRef:map[apiGroup:rbac.authorization.k8s.io kind:ClusterRole name:system:node-proxier] subjects:[map[kind:ServiceAccount name:kube-proxy namespace:kube-system]]]})]" should have 3 item(s), but has 6
#166 15.06         	Test:       	TestManifestSuite/TestReconcileKubeProxyExtraArgs
#166 15.06     manifest_test.go:210: tear down
#166 15.06 controller-runtime:  k8s.ManifestController: 2021/04/12 21:19:56 controller finished
#166 15.06 --- FAIL: TestManifestSuite (0.32s)
#166 15.06     --- PASS: TestManifestSuite/TestReconcileDefaults (0.10s)
#166 15.06     --- PASS: TestManifestSuite/TestReconcileDisableKubeProxy (0.10s)
#166 15.06     --- FAIL: TestManifestSuite/TestReconcileKubeProxyExtraArgs (0.10s)
#166 15.06 FAIL
#166 15.06 FAIL	github.com/talos-systems/talos/internal/app/machined/pkg/controllers/k8s
```

### Environment

- Talos version: [`talosctl version --nodes <problematic nodes>`]
- Kubernetes version: [`kubectl version --short`]
- Platform:

--- a/pkg/resources/k8s/manifest.go
+++ b/pkg/resources/k8s/manifest.go
@@ -93,6 +93,7 @@ func (r *Manifest) ResourceDefinition() meta.ResourceDefinitionSpec {
 
 // SetYAML parses manifest from YAML.
 func (r *Manifest) SetYAML(yamlBytes []byte) error {
+	r.spec.Items = nil
 	reader := yaml.NewYAMLReader(bufio.NewReader(bytes.NewReader(yamlBytes)))
 
 	for {

